app/controllers/contributions_controller.rb

```diff
@@ -1,5 +1,30 @@
 class ContributionsController < ApplicationController
+  before_action :authenticate_user!
+
   def new
-    
+    @project = get_project
+    @contribution = Contribution.new(user: current_user, project: @project)
+  end
+  
+  def create
+    @project = get_project
+    result = Contributions::CreateTransaction.call(contribution_params.merge(project: @project, user: current_user))
+    if result.success
+      redirect_to users_dashboards_path
+    else
+      @contribution = result.failure[:contribution]
+      @counterparts = set_counterparts
+      render :new
+    end
+  end
+
+  private
+
+  def get_project
+    Project.find(params[:project_id]).decorate
+  end
```

Comment:
relis cette partie : https://github.com/drapergem/draper#when-to-decorate-objects

Reply:
Oui je me souviens mais là le projet n'est pas la resource du controller donc ça ne fonctionne pas le `decorates_assigned`

Reply:
Non, ce que ça dit, c'est de ne pas utiliser les objets décorés dans le controller, ce que tu fais actuellement.

Reply:
Ah ok tu veux dire le décorer juste avant de l'envoyer à la vue, et là je le fais trop tôt.